ssl/statem/statem_clnt.c

```diff
@@ -821,7 +821,10 @@ int tls_construct_client_hello(SSL *s)
         goto err;
     }
 
-    if ((sess == NULL) || (sess->ssl_version != s->version) ||
+    if ((sess == NULL) ||
+        /* Only destroy session which version is bigger than maximum supported.
+         */
+        ssl_version_cmp(s, sess->ssl_version, s->version) > 0 ||
```

Comment:
Oh right, that was different function. Sorry!  Should it just check for `any` methods here, then? And if method is not one of `any`, do exact version comparison (`!= 0`)